backend/services/detailed_reports_service.py

```
@@ -0,0 +1,334 @@
+import os
+
+class DetailedReportsService:
+    def __init__(self):
+        self.astro_calculator = AstroCalculator()
+        self.claude_ai = ClaudeAI()
+        # In production, use Redis for report storage
+        self.redis_client = None
+        try:
+            if os.environ.get('REDIS_URL'):
+                self.redis_client = redis.from_url(os.environ.get('REDIS_URL'))
+        except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)